src/lib/ui/ExtensionIcon.svelte

```diff
@@ -0,0 +1,21 @@
+<script lang="ts">
+import type { CatalogExtensionInfo } from '../api/extensions-info';
+
+const { extension, size = 'normal' }: { extension: CatalogExtensionInfo; size?: 'normal' | 'xl' } = $props();
+
+const iconUrl: string | undefined = $derived.by(() => {
+  // take the icon of the latest version
+  return extension.versions?.[0]?.files.find(file => file.assetType === 'icon')?.data ?? '';
+});
+</script>
+
+{#if iconUrl}
+	<img
+		src={iconUrl}
```

Comment:
There should be default extension icon, for the case when icon is not provided.

Reply:
it is mandatory to be provided